ns.go

```diff
@@ -0,0 +1,208 @@
+		}
+
+		creds := credentials.NewClientTLSFromCert(pool, "")
+		o.gopts = append(o.gopts, grpc.WithTransportCredentials(creds))
+		return nil
+	}
+}
+
+// WithDgraphAPIKey will use the provided API key for authentication for Dgraph Cloud.
+func WithDgraphAPIKey(apiKey string) ClientOption {
+	return func(o *clientOptions) error {
+		if err := withSystemCertPool()(o); err != nil {
+			return err
+		}
+
+		o.gopts = append(o.gopts, grpc.WithPerRPCCredentials(&authCreds{token: apiKey}))
+		return nil
+	}
+}
+
+// WithBearerToken uses the provided token and presents it as a Bearer Token
+// in the HTTP Authorization header for authentication against a Dgraph Cluster.
+// This can be used to connect to Hypermode Cloud.
+func WithBearerToken(token string) ClientOption {
+	return func(o *clientOptions) error {
+		if err := withSystemCertPool()(o); err != nil {
+			return err
+		}
+
+		o.gopts = append(o.gopts, grpc.WithPerRPCCredentials(&bearerCreds{token: token}))
+		return nil
+	}
+}
+
+// WithGrpcOption will add a grpc.DialOption to the client.
+// This is useful for setting custom  grpc options.
+func WithGrpcOption(opt grpc.DialOption) ClientOption {
+	return func(o *clientOptions) error {
+		o.gopts = append(o.gopts, opt)
+		return nil
+	}
+}
+
+// NewClient creates a new Dgraph client for a single endpoint.
+func NewClient(endpoint string, opts ...ClientOption) (*Dgraph, error) {
+	return NewRoundRobinClient([]string{endpoint}, opts...)
+}
+
+// NewRoundRobinClient creates a new Dgraph client for a list
+// of endpoints. It will round robin among the provided endpoints.
+func NewRoundRobinClient(endpoints []string, opts ...ClientOption) (*Dgraph, error) {
+	co := &clientOptions{}
+	for _, opt := range opts {
+		if err := opt(co); err != nil {
+			return nil, err
+		}
+	}
+
+	conns := make([]*grpc.ClientConn, len(endpoints))
+	dc := make([]api.DgraphClient, len(endpoints))
+	dcv25 := make([]apiv25.DgraphHMClient, len(endpoints))
+	for i, endpoint := range endpoints {
+		conn, err := grpc.NewClient(endpoint, co.gopts...)
+		if err != nil {
+			return nil, fmt.Errorf("failed to connect to endpoint [%s]: %w", endpoint, err)
+		}
+		conns[i] = conn
+		dc[i] = api.NewDgraphClient(conn)
+		dcv25[i] = apiv25.NewDgraphHMClient(conn)
+	}
+
+	return &Dgraph{dc: dc, dcv25: dcv25}, nil
+}
+
+// Close closes all the connections to the Dgraph Cluster.
+func (d *Dgraph) Close() {
+	for _, conn := range d.conns {
+		_ = conn.Close()
+	}
+}
+
+func (d *Dgraph) anyClientv25() apiv25.DgraphHMClient {
+	//nolint:gosec
+	return d.dcv25[rand.Intn(len(d.dcv25))]
+}
+
+// CreateNamespace creates a new namespace with the given name and password for groot user.
+func (d *Dgraph) CreateNamespace(ctx context.Context, name string, password string) error {
+	dc := d.anyClientv25()
+	req := &apiv25.CreateNamespaceRequest{NsName: name, GrootPassword: password}
+	_, err := dc.CreateNamespace(d.getContext(ctx), req)
+	if isJwtExpired(err) {
```

Comment:
I don't like that after every time we do something, we are doing isJwtExpired. Can we write a middleware somewhere?